provider/pkg/await/await.go

```diff
@@ -536,13 +536,22 @@ func csaUpdate(c *UpdateConfig, liveOldObj *unstructured.Unstructured, client dy
 	return client.Patch(c.Context, liveOldObj.GetName(), patchType, patch, options)
 }
 
+type patcher interface {
+	Patch(ctx context.Context, name string, pt types.PatchType, data []byte, options metav1.PatchOptions, subresources ...string) (*unstructured.Unstructured, error)
+}
+
 // ssaUpdate handles the logic for updating a resource using server-side apply.
-func ssaUpdate(c *UpdateConfig, liveOldObj *unstructured.Unstructured, client dynamic.ResourceInterface) (*unstructured.Unstructured, error) {
+func ssaUpdate(c *UpdateConfig, liveOldObj *unstructured.Unstructured, client patcher) (*unstructured.Unstructured, error) {
 	liveOldObj, err := fixCSAFieldManagers(c, liveOldObj, client)
 	if err != nil {
 		return nil, err
 	}
 
+	fieldManager := c.FieldManager
```

Comment:
Looks like this is deadcode from testing? Let's drop the `fieldManager` changes in this function.

Reply:
Yeah, this is showing where we could impersonate `kubectl` if we wanted to go down that route, but it seems like we're OK forcing the patch.